utils/tests/unit/test_test_results.py

```diff
@@ -0,0 +1,144 @@
+import datetime as dt
+
+from shared.django_apps.reports.models import TestInstance
+
+from utils.test_results import aggregate_test_results
+
+
+class TestSyncTestResults:
+    def test_sync_test_results(
+        self,
+        transactional_db,
+        repo_fixture,
+        create_test_func,
+        create_test_instance_func,
+    ):
+        test_1 = create_test_func()
+        test_2 = create_test_func()
+
+        create_test_instance_func(test_1, TestInstance.Outcome.FAILURE, "a", duration=1)
+        create_test_instance_func(test_1, TestInstance.Outcome.PASS, "b", duration=2)
+        create_test_instance_func(test_2, TestInstance.Outcome.SKIP, "c", duration=1)
+        create_test_instance_func(test_2, TestInstance.Outcome.ERROR, "d", duration=1)
+        create_test_instance_func(test_2, TestInstance.Outcome.PASS, "d", duration=1)
+        create_test_instance_func(test_2, TestInstance.Outcome.FAILURE, "d", duration=1)
+        create_test_instance_func(test_2, TestInstance.Outcome.FAILURE, "d", duration=1)
+        create_test_instance_func(test_2, TestInstance.Outcome.ERROR, "e", duration=1)
+
+        d = aggregate_test_results(repo_fixture.repoid)
+
+        assert d[test_1.id]["failure_rate"] == 0.5
```

Comment:
can we have an "expected" dict and just compare both dicts? It'd be easier to understand what we actually expect :E

It also has the (small) added benefit to check that there's nothing extra that we don't expect in the response